### ISSUE TITLE
Make wildfly sender wait until other subsystems are up

### ISSUE THREAD

**Issue by pilhuhn** [hawkular/wildfly-monitor#4]
[ Migrated from https://github.com/rhq-project/rhq-metrics/issues/25 ]

Currently it is possible that on wildfly console you see something like

07:12:17,288 ERROR [stderr](RHQ-Metrics-Sender-Thread)     ("type" => "threading")
07:12:17,289 ERROR [stderr](RHQ-Metrics-Sender-Thread) ]', attribute='thread-count', type='UNKNOWN'} not known
07:12:17,291 ERROR [stderr](RHQ-Metrics-Sender-Thread)         at org.rhq.metrics.clients.wflySender.service.RhqMetricsService.obtainMetricValue(RhqMetricsService.java:188)
07:12:17,292 ERROR [stderr](RHQ-Metrics-Sender-Thread)         at org.rhq.metrics.clients.wflySender.service.RhqMetricsService.access$
This comes from the fact that at the moment the metric definition from <metric> tag of the subsystem declaration is evaluated, the subsystem from where to take the metric is not yet fully up and thus can not yet answer questions about the contents of its model.

We need to either make sure we wait long enough on startup. Or re-evaluate failed ones from time to time until they are good.

@heiko-braun wrote:
This can probably be resolved by declaring a service dependency, like

```
    ServiceController<CassandraService> controller = context.getServiceTarget()
            .addService(SERVICE_NAME, service)
            .addListener(verificationHandler)
            .setInitialMode(ServiceController.Mode.ACTIVE)
            .addDependency(PathManagerService.SERVICE_NAME, PathManager.class, service.getPathManagerInjector())
            .install();
```

we just need to figure what that dependency actually is to postpone the metric startup until all subsystems are there.

In the example above the CassandraService waits for the PathManager.

@pilhuhn wrote: 

Unfortunately the list of objects to monitor is not known at compile time.

Wfly 8.1 displays this at the end of startup:

09:57:19,036 INFO  [org.jboss.as](Controller Boot Thread) JBAS015961: Http management interface listening on http://127.0.0.1:9990/management
09:57:19,044 INFO  [org.jboss.as](Controller Boot Thread) JBAS015951: Admin console listening on http://127.0.0.1:9990
09:57:19,045 INFO  [org.jboss.as](Controller Boot Thread) JBAS015874: WildFly 8.1.0.Final "Kenny" started in 36824ms - Started 349 of 408 services (101 services are lazy, passive or on-demand)
Perhaps we can wait on the "started in .. " message. Or that the management interface is finally listening.

@heiko-braun wrote: 

[15:46] Emmanuel Hugonnet: heiko then why don't you wait until boot is done before having your server running 'effectively' ?
[15:46] Heiko Braun: and somehow listen to that?
that's fine as well
[15:47] Emmanuel Hugonnet: ther is a service for that
[15:47] Heiko Braun: cool
any pointers?
[15:47] Emmanuel Hugonnet: yes
[15:47] Emmanuel Hugonnet: give me 5 secs
[15:49] Emmanuel Hugonnet: ControlledProcessStateService
it is marked as deprecated
you can register your service as a listner
to it also
[15:51] Tomaz Cerar: service listeners are deprecated as they dont work properly
we moved almost all code off it
